src/Yolo.java

```diff
@@ -94,42 +92,38 @@ private static void analyzeFrame(VideoCapture cap, Mat frame, Net net) {
 		int height;
 		int left;
 		int top;
+		Mat row;
+		Mat scores;
+		Mat level;
+		Core.MinMaxLocResult mm;
+		float confidence;
+		Point classIdPoint;
 		
-		/*clsIds.clear();
-		confs.clear();
-		rects.clear();*/
-
-		Mat blob = Dnn.blobFromImage(frame, 0.00392, sz, new Scalar(0), true); 
+		Mat blob = Dnn.blobFromImage(frame, 0.00392, sz, new Scalar(0), true);  //edit this maybe, scalar is empty rn, so no mean subtraction i think
 		net.setInput(blob);
-		net.forward(result, outBlobNames);
-
-		//outBlobNames.forEach(System.out::println);
-		// result.forEach(System.out::println);
+		net.forward(result, outBlobNames); //result is a 4d tensor, images, height, width, color channels
 
 		float confThreshold = 0.6f; // Insert thresholding beyond which the model will detect objects//
 
 		for (int i = 0; i < result.size(); ++i) {
 			// each row is a candidate detection, the 1st 4 numbers are
 			// [center_x, center_y, width, height], followed by (N-4) class probabilities
-			Mat level = result.get(i);
-			Mat row;// = level.row(j);
-			Mat scores;// = row.colRange(5, level.cols());
-			Core.MinMaxLocResult mm;// = Core.minMaxLoc(scores);
-			float confidence;// = (float) mm.maxVal;
-			Point classIdPoint;// = mm.maxLoc;
+			
+			level = result.get(i); //gets i output blob image from network, now a 3d tensor, height, width, color channels
+			
 			for (int j = 0; j < level.rows(); ++j) {
-				row = level.row(j);
-				scores = row.colRange(5, level.cols());
-				mm = Core.minMaxLoc(scores);
-				confidence = (float) mm.maxVal;
-				classIdPoint = mm.maxLoc;
+				row = level.row(j); // gets the data for the image, 
+				scores = row.colRange(5, level.cols()); //scores are class probabilities listed after the first 4 values
+				mm = Core.minMaxLoc(scores); //finds maximum score in class probabilities
+				confidence = (float) mm.maxVal; //confidence = maxVal
+				classIdPoint = mm.maxLoc; //Id index is the index of maxVal
+				
 				if (confidence > confThreshold) {
-					centerX = (int) (row.get(0, 0)[0] * frame.cols()); // scaling for drawing the bounding
-																		// boxes//
-					centerY = (int) (row.get(0, 1)[0] * frame.rows());
-					width = (int) (row.get(0, 2)[0] * frame.cols());
-					height = (int) (row.get(0, 3)[0] * frame.rows());
-					left = centerX - width / 2;
+					centerX = (int) (row.get(0, 0)[0] * frame.cols()); //gets centerX from output blob and scales it for the input image
+					centerY = (int) (row.get(0, 1)[0] * frame.rows()); //same but with centerY
+					width = (int) (row.get(0, 2)[0] * frame.cols()); //same but with width
+					height = (int) (row.get(0, 3)[0] * frame.rows());//same but with height
```

Comment:
Should we use float/double for these values?

Reply:
No need, because these are pixel coordinates and pixel coordinates are integers. Also, the multiplication cannot result in a decimal.